File: backend/models/Report.js
import mongoose from 'mongoose';

const ReportSchema = new mongoose.Schema({
  title: { type: String, required: true },
  description: String,
  category: { type: String, enum: ['Sampah','Air','Udara','Fasilitas'], required: true },
  photos: [String], // path relatif, mis. "/uploads/abc.jpg"
  location: {
    type: { type: String, enum: ['Point'], default: 'Point' },
    coordinates: { type: [Number], required: true } // [lng, lat]
  },
  status: { type: String, enum: ['Menunggu','Diterima','Diproses','Selesai','Ditolak'], default: 'Menunggu' },
  history: [{ status: String, note: String, at: Date, by: String }]
}, { timestamps: true });

ReportSchema.index({ location: '2dsphere' });

export default mongoose.model('Report', ReportSchema);

File: sorot_app/lib/widgets/report.dart
class Report {
  final String id;
  final String title;
  final String description;
  final String status;
  final double lat;
  final double lng;

  Report({
    required this.id,
    required this.title,
    required this.description,
    required this.status,
    required this.lat,
    required this.lng,
  });

  factory Report.fromJson(Map<String, dynamic> json) {
    return Report(
      id: json['_id'] ?? '',
      title: json['title'] ?? '',
      description: json['description'] ?? '',
      status: json['status'] ?? 'pending',
      lat: (json['lat'] ?? 0).toDouble(),
      lng: (json['lng'] ?? 0).toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'title': title,
      'description': description,
      'status': status,
      'lat': lat,
      'lng': lng,
    };
  }
}
